example/client.js

```diff
@@ -1,11 +1,17 @@
 // @flow
 import createClient from '../src';
 
-
 const client = createClient('SimpleStepPublisher', 'A simple chorus step publisher');
 export default client;
 
-export const clientOpened: Promise<void> =
+// Detect whether chorus was enabled by setting a http param in the URL when loading the app
+const urlParams = new URLSearchParams(window.location.search);
+const chorusEnabled = urlParams.get('chorusEnabled');
+
+// If Chorus was enabled we need to open a connection to the Chrous interpreter which
+// is running in the container named 'chorus-interpreter' on port 9080
+export const clientOpened : Promise<void> = chorusEnabled ?
 	client
-		.open('ws://localhost:9080')
-		.then(() => client.connect());
+		.open('ws://chorus-interpreter:9080')
+		.then(() => client.connect()) :
+	new Promise(() => {});
```

Comment:
in the case chorus is not enable, did you mean to reject the promise?
If so, you can write: `Promise.reject()`, or if you do need to resolve to nothing like you do here, yu can write as `Promise.resolve()`

Reply:
I was intending to leave the Promise pending - although that may not be a good approach. Resolving the promise would trigger step publication, which would fail since we have no connection, so my first concern was not to resolve it. But probably leaving a promise pending is not very idiomatic, so perhaps it would be better to reject it with a warning message to show that chorus was not enabled, e.g reject('Chorus connection was not enabled'). I'll make that change if you agree.

Reply:
👍 Definitely the right way.